fastly/resource_fastly_ngwaf_thresholds.go

```diff
@@ -177,7 +177,7 @@ func resourceFastlyNGWAFThresholdsUpdate(ctx context.Context, d *schema.Resource
 
 	log.Printf("[DEBUG] UPDATE: NGWAF threshold input: %#v", i)
 
-	_, err := wsr.Update(gofastly.NewContextForResourceID(ctx, d.Id()), conn, &i)
+	_, err := wsr.Update(gofastly.NewContextForResourceID(ctx, d.Get("workspace_id").(string)), conn, &i)
```

Comment:
For consistency this should use the same technique as the other functions, extracting the workspaceID just once and using it in all of the places where it is needed.